src/Components/WebAssembly/WebAssembly/src/Hosting/WebAssemblyHostConfiguration.cs

```diff
@@ -0,0 +1,187 @@
+        private ConfigurationReloadToken _changeToken = new ConfigurationReloadToken();
+
+        /// <summary>
+        /// Returns the sources used to obtain configuration values.
+        /// </summary>
+        public IList<IConfigurationSource> Sources => _sources.ToList();
+
+        /// <summary>
+        /// Returns the providers user to obtain configuration values.
+        /// </summary>
+        public IEnumerable<IConfigurationProvider> Providers => _providers.ToList();
+
+        /// <summary>
+        /// Gets a key/value collection that can be used to share data between the <see cref="IConfigurationBuilder"/>
+        /// and the registered <see cref="IConfigurationProvider"/>s.
+        ///
+        /// In this implementation, this largely exists as a way to satisfy the
+        /// requirements of the IConfigurationBuilder and is not populated by
+        /// <see cref="WebAssemblyHostConfiguration"/> with any meaningful info.
+        /// </summary>
+        IDictionary<string, object> IConfigurationBuilder.Properties { get; } = new Dictionary<string, object>();
+
+        /// <inheritdoc />
+        public string this[string key]
+        {
+            get {
+                // Iterate through the providers in reverse to extract
+                // the value from the most recently inserted provider.
+                for (var i = _providers.Count - 1; i >= 0; i--)
+                {
+                    var provider = _providers[i];
+
+                    if (provider.TryGet(key, out var value))
+                    {
+                        return value;
+                    }
+                }
+
+                return null;
+            }
+            set {
+                if (_providers.Count == 0)
+                {
+                    throw new InvalidOperationException("Can only set property if at least one provider has been inserted.");
+                }
+
+                foreach (var provider in _providers)
+                {
+                    provider.Set(key, value);
+                }
+
+            }
+        }
+
+        /// <summary>
+        /// Gets a configuration sub-section with the specified key.
+        /// </summary>
+        /// <param name="key">The key of the configuration section.</param>
+        /// <returns>The <see cref="IConfigurationSection"/>.</returns>
+        /// <remarks>
+        ///     This method will never return <c>null</c>. If no matching sub-section is found with the specified key,
+        ///     an empty <see cref="IConfigurationSection"/> will be returned.
+        /// </remarks>
+        public IConfigurationSection GetSection(string key) => new ConfigurationSection(this, key);
+
+        /// <summary>
+        /// Gets the immediate descendant configuration sub-sections.
+        /// </summary>
+        /// <returns>The configuration sub-sections.</returns>
+        public IEnumerable<IConfigurationSection> GetChildren()
```

Comment:
I'm not super sure how commonly this API is used. Perhaps this could be an explicit implementation? @anurse thoughts \ suggestions?

Reply:
What's the context? The API `IConfiguration.GetChildren` is certainly used by some components.

Reply:
Yeah, I've used this API in the past, the implementation for that should be public

Reply:
IMO, I actually think this should be an explicit implementation. I don't see a scenario where the user would want to call `GetChildren()` directly on `WebAssemblyHostConfiguration`.  It might also be sensible to start with a more restrictive API then open it up if that is something users express interested in it.

Reply:
I guess I'm not clear why you'd want to **hide** this. It's something a user would *expect* to see when accessing an implementation of `IConfiguration` or something that looks like one.

Does it not work as expected in this scenario? I'm not versed in the context for the change at large though.

Reply:
This change introduces a new class that implements both the IConfiguration and IConfigurationBuilder interfaces. Making it an explicit implementation requires that the user be accessing the object as an `IConfiguartion` which is the behavior that we want to encourage.

Reply:
Ah, ok. That makes a lot of sense then :). You'd probably also want to do it for things like `GetSection` and the indexer then.